### PR TITLE
feat: add option to enable usd wallets globally

### PR DESCRIPTION
**Pull request by vindard**
## Description

This PR adds a config option that allows an instance owner to toggle whether USD wallets are also added for each new user created.

### CODE REVIEW

**Review by nicolasburtey — COMMENTED** [review]
One suggestion on how to "architecture" this

File: default.yaml [hunk 1]
```diff
@@ -130,6 +130,7 @@ rateLimits:
 
 accounts:
   initialStatus: active
+  hasUsdWallet: false
```
[the remark this answers is not in the document]
Comment (vindard) [reply default.yaml:133]: Oh yea I like this. I've already had folks asking if we could run our stack with USD-only.

File: src/app/users/create-user.ts [hunk 2]
```diff
@@ -10,21 +9,45 @@ import {
...
+    enabledWallets[currency] = wallet
+  }
 
-  account.defaultWalletId = wallet.id
+  // Set default wallet explicitly, or implicitly as 1st element in
+  // walletsEnabledConfig array.
+  const defaultWalletId =
+    enabledWallets[defaultWalletConfig]?.id || enabledWallets[walletsEnabledConfig[0]]?.id
+
+  if (defaultWalletId === undefined) {
```
Comment (nicolasburtey) [inline src/app/users/create-user.ts:47]: I think that is an acceptable use case to not have a defaultWalletId ?

like for API/business users (when we eventually get there), you create your account, but you don't necessarily have to have a default wallet at the start.

that said, maybe it's possible our backend doesn't handle yet this use case (?). In that case I would recommand still making it optional to not have a default wallet from the config view point, and we can dive more into the backend if there are some edges cases to fix when the need arise

File: src/app/users/login.ts [hunk 3]
```diff
@@ -60,7 +61,10 @@ export const login = async ({
   if (user instanceof CouldNotFindUserFromPhoneError) {
     subLogger.info({ phone }, "new user signup")
     const userRaw: NewUserInfo = { phone }
-    user = await createUserForPhoneSchema(userRaw)
+    user = await createUserForPhoneSchema({
+      newUserInfo: userRaw,
+      config: getAccountsConfig(),
```
Comment (nicolasburtey) [inline src/app/users/login.ts:66]: I would rename the function more into something like `getDefaultAccountsConfig`.

the idea is that account will be able to evolve over time. ie: as a enterprise user, I may want to add new wallets to my account.

File: src/config/schema.ts [hunk 4]
```diff
@@ -219,8 +220,26 @@ export const configSchema = {
       type: "object",
       properties: {
         initialStatus: { type: "string", enum: Object.values(AccountStatus) },
+        wallets: {
```
Comment (nicolasburtey) [inline src/config/schema.ts:223]: suggestion: defaultWallets or initialWallets

**Review by nicolasburtey — COMMENTED** [review]
some suggestions on the wording, and also the behavior when the initial wallet set is empty, but otherwise looks good.

**Comment by vindard** [comment]
> some suggestions on the wording, and also the behavior when the initial wallet set is empty, but otherwise looks good. 

Made some changes based on suggestions.

**Review by nicolasburtey — APPROVED** [review]

**Pull request merged**